app/code/core/Mage/Core/Model/Url.php

```diff
@@ -868,6 +868,7 @@ public function setQueryParams(array $data)
      */
     public function purgeQueryParams()
     {
+        $this->unsetData('query');
         $this->setData('query_params', []);
```

Comment:
since the `$this->setData('query_params', [])` would it be wrong to `$this->unsetData('query_params')`?

Reply:
You're right, fixed. I had though the model always expected query_params to be an array, but it seems all the right checks are already there.